packages/next/src/build/webpack/plugins/flight-client-entry-plugin.ts

```diff
@@ -196,7 +196,7 @@ export class FlightClientEntryPlugin {
         // so it's only necessary to add it for matchResource or mod.resourceResolveData
         const modResource = modPath
           ? modPath.startsWith(BARREL_OPTIMIZATION_PREFIX)
-            ? mod.resource
+            ? mod.resource + '@' + modPath
```

Comment:
We could probably create a helper for this case and use that everywhere got creating a modified resource, and commen there about `BARREL_OPTIMIZATION_PREFIX`. Not blocking